### PR TITLE
Make sign up invite only

### PR DESCRIPTION
**Pull request by domwong**
Block sending of initial email in signup flow if they are not in the invite service

### CODE REVIEW

File: signup/handler/signup.go [hunk 1]
```diff
@@ -127,7 +136,8 @@ func (e *Signup) SendVerificationEmail(ctx context.Context,
 	}
 
 	if e.testMode {
-		logger.Infof("Sending verification token '%v'", k)
+		logger.Infof("Test mode enabled, skipping send. Verification token is '%v'", k)
+		return nil
```
Comment (crufter) [inline signup/handler/signup.go:140]: I think we should not return and send the email, token logging is only here not having to read the email.

**Comment by crufter** [comment]
Before merging this can we make sure the micro m3o signup test still works?

File: signup/handler/signup.go (hunk 1 above)
Comment (domwong) [reply signup/handler/signup.go:140]: ah ok. misunderstood the point of it. will fix

Comment (crufter) [reply signup/handler/signup.go]: No worries, not an important point really. Cheers

**Comment by domwong** [comment]
> Before merging this can we make sure the micro m3o signup test still works? ie. this https://github.com/micro/micro/blob/master/test/m3o_test.go#L24

Have lined up a PR for updating the tests since we now need to run invite service too.

**Pull request merged**